app/src/main/java/com/easyfitness/machines/MachineFragment.java

```diff
@@ -222,7 +271,22 @@ private void refreshData() {
             if (getProfil() != null) {
 
                 // Version avec table Machine
-                c = mDbMachine.getAllMachines();
+                String requiredTypes = "(";
+                int numRequiredTypes = 0;
+
+                for (int i = 0; i < checkedFilterItems.length; i++) {
+                    if (checkedFilterItems[i]) {
+                        requiredTypes = requiredTypes.concat(i + ",");
+                        numRequiredTypes++;
+                    }
+                }
+                if (numRequiredTypes == 0) {
+                    requiredTypes = requiredTypes.concat(")");
+                } else {
+                    requiredTypes = requiredTypes.substring(0, requiredTypes.length() - 1).concat(")");
+                }
```

Comment:
I think the string builder should rather be in the DAO class as you are creating some part of the SQL request here. 
And instead of passing an array of integers that have no direct meaning, it would be preferable to pass an array of **ExerciseType** enum values.

Reply:
You are definitely right with moving the string builder to the DAO class. I did that. 

However, I am not sure about passing an array of **ExerciseTypes** since the data structure which holds the information about the `checkedFilterItems` is defined as boolean[].  That's why I changed the implementation to pass the array `checkedFilterItems`, since I thought the translation to another data structure would not add significant value. What do you think about the current implementation?

Reply:
Thanks for moving the code to DAO. 
Concerning my second request, the idea is to make the code easier to understand and more future proof. What you did works perfectly fine but looking at the getRequiredTypesAsString function, you have to know that the first boolean is for strength, the second for cardio and the third for iso. And also your function only works if the checklist is in that order. The idea would to replace the meaningless boolean array from the checklist to ExerciseType array as close as possible of the checklist (in the AlertDialog return for example). 
In term of performance, my proposal is surely less efficient but as the filter is displayed only on demand and is only one object, we don't really care.
Let me know if you disagree.
Thanks.

Reply:
The problem with the boolean array should be solved within b6b797873db800859fa1bb7c2a19314f6cfee467. Let me know if there are more changes required.

Reply:
That's perfect! Thank you for the update.